map/src/map/util/Clusterizer.js

```diff
@@ -11,29 +12,114 @@ export function clusterMarkers({
     iconSize = EXPLORE_BIG_ICON_SIZE,
     secondaryIconSize = 10,
     isPoi = false,
+    isFavorites = false,
 }) {
-    const maxMainPlaces = isPoi ? 2000 : 50;
-    const maxSecondaryPlaces = zoom > 10 ? 200 : 900;
-    const useUniformMarkerPlacement = isPoi ?? (zoom <= 10 || zoom >= 16);
+    const maxMainPlaces = getMaxMainPlaces(zoom, isPoi, isFavorites);
+    const maxSecondaryPlaces = getMaxSecondaryPlaces(zoom, isFavorites);
+    const useUniformMarkerPlacement = getUseUniformMarkerPlacement(zoom, isPoi, isFavorites);
 
-    const shift = 1; // Adjust shift as needed for your specific case
-    const clustered = {};
+    // Minimum distances between markers in meters
+    const mainMinDistance = calculateDistance({
+        iconSize,
+        latitude,
+        zoom,
+        isPoi,
+    });
+
+    const secondaryMinDistance = calculateDistance({
+        iconSize: secondaryIconSize,
+        latitude,
+        zoom,
+        isPoi,
+    });
+
+    // Sort clusters by size
+    const clusters = Object.values(clusterPlaces(places, zoom, isFavorites));
+
+    const mainMarkers = createMainMarkersArr(clusters, useUniformMarkerPlacement, mainMinDistance, isFavorites);
 
-    // Function to convert pixels to meters
-    const metersPerPixel = (latitude, zoomLevel) => {
-        const earthCircumference = 40075017; // In meters
-        return (earthCircumference * Math.cos((latitude * Math.PI) / 180)) / Math.pow(2, zoomLevel + 8);
+    const secondaryMarkers = createOtherMarkersArr(
+        clusters,
+        mainMarkers,
+        maxMainPlaces,
+        mainMinDistance,
+        secondaryMinDistance,
+        isFavorites
+    );
```

Comment:
Please use destucting-style for such a big number of parameters:

{
        clusters,
        mainMarkers,
        maxMainPlaces,
        mainMinDistance,
        secondaryMinDistance,
        isFavorites
}